rust/lance-index/src/vector/ivf.rs

```diff
@@ -463,23 +462,11 @@ impl<T: ArrowFloatType + Dot + L2 + Cosine + 'static> Ivf for IvfImpl<T> {
             (Some(partitions), Some(row_ids)) => {
                 debug!("Using precomputed partitions for partitions");
                 let mut builder = UInt32Builder::new();
-                for row in row_ids
-                    .as_any()
-                    .downcast_ref::<UInt64Array>()
-                    .unwrap()
-                    .values()
-                    .iter()
-                {
+                for row in row_ids.as_primitive::<UInt64Type>().values().iter() {
                     if let Some(part_id) = partitions.get(row) {
                         builder.append_value(*part_id);
                     } else {
-                        return Err(Error::Index {
-                            message: format!(
-                                "Row ID {} does not exist in the precomputed partitions",
-                                row
-                            ),
-                            location: location!(),
-                        });
+                        builder.append_null();
```

Comment:
do we need to handle the null in the shuffle?

Reply:
These null partitions will be filtered out.